public/app/features/panel/components/PanelRenderer.tsx

```diff
@@ -128,6 +138,6 @@ function useFieldOverrides(
   }, [fieldConfigRegistry, fieldConfig, data, series, timeZone, theme]);
 }
 
-function pluginIsLoading(loading: boolean, plugin: PanelPlugin<any, any> | undefined, pluginId: string) {
-  return loading || plugin?.meta.id !== pluginId;
+function pluginHasChanged(plugin: PanelPlugin | undefined, pluginId: string) {
+  return plugin && plugin.meta.id !== pluginId;
 }
```

Comment:
```suggestion
function hasPluginChanged(plugin: PanelPlugin | undefined, pluginId: string) {
  return plugin && plugin.meta.id !== pluginId;
}
```

Nit - can we move it outside of PanelRendered and expose as an util? I would use it here https://github.com/grafana/grafana/blob/main/public/app/features/query/state/updateQueries.ts#L14 :)

Reply:
@dprokop 
moved it to a instance function on PanelPlugin, made the code a bit cleaner